vim: Fix the goto definition action in vim where it goes into visual mode

shaik-zeeshan opened the pull request:
Closes #42596

fixes vim going into visual mode whenever the goto definition action is done

Release Notes:

- N/A

shaik-zeeshan committed:
fixes the goto definition in vim where it goes into visual mode

shaik-zeeshan commented:
@cla-bot check

terids commented:
@dinocosta  I can reproduce the issue 100% when working with C, but not with Swift. Might be language specific.

dinocosta committed:
refactor(editor): collapse range when vim mode is enabled

Revert the original changes done to fix the issue with `g d` in vim mode
updating the mode to Visual.

After looking into this it appears this was accidentally changed in
https://github.com/zed-industries/zed/pull/41583 with the removal of the
`collapse_matches` field from the `Editor`, which would be provided to
the `range_for_match` method and, in the case where vim mode was
enabled, it was `true`, avoiding the change to visual mode when jumping
to definition.

This commit updates the `Editor.navigate_to_hover_links` method in order
to check if vim mode is enabled, in which case it now collapses the
range, otherwise it won't, returning to the previous behavior.
